### PR TITLE
Support to scoped names

### PR DESCRIPTION
**Pull request by richiware**
Added support for scoped names. Also, Modules are moved to IDL parser/generator.

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/xtypes-manual-linux/119/)

File: examples/iterators.cpp [hunk 1]
```diff
@@ -13,19 +12,19 @@ int main()
         DynamicData str1(string);
 
         str1.value<std::string>("Hello!");
-        for (ReadableDynamicDataRef&& elem : str1)
+        for (ReadableDynamicDataRef && elem : str1)
```
Comment (LuisGP) [inline examples/iterators.cpp]:
```suggestion
        for (ReadableDynamicDataRef&& elem : str1)
```

Comment (LuisGP) [reply examples/iterators.cpp]: In all PR.

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/xtypes-manual-linux/120/)

**Review by LuisGP — DISMISSED** [review]

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/xtypes-manual-linux/121/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/xtypes-manual-linux/122/)

File: examples/iterators.cpp [hunk 2]
```diff
@@ -13,19 +12,19 @@ int main()
         DynamicData str1(string);
 
         str1.value<std::string>("Hello!");
-        for (ReadableDynamicDataRef&& elem : str1)
+        for (auto&& elem : str1)
```
Comment (LuisGP) [inline examples/iterators.cpp]: This is an example, using `auto` obfuscates the code making it more difficult to understand.

**Review by LuisGP — CHANGES_REQUESTED** [review]
Avoid `auto`.

**Review by LuisGP — APPROVED** [review]

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/xtypes-manual-linux/123/)

**Pull request merged**